ci/cppcheck/cppcheck-new-issues.sh

```diff
@@ -25,8 +25,19 @@ do
     if ! grep -q -F "$i" "$rootdir"/ci/cppcheck/cppcheck_existing_issues.txt ; then
         status=1
         err "New issue:"
-        grep -F "$i" "$rootdir/cppcheck_results.txt"
+        "$i"
     fi
 done
 
+# Generate a new list of existing issues by removing the ones that are now solved
+comm -12 <(sort "$rootdir"/cppcheck_results.txt) <(sort "$rootdir"/ci/cppcheck/cppcheck_existing_issues.txt)  > "$rootdir"/cppcheck_existing_issues.txt
+
+mv "$rootdir"/cppcheck_existing_issues.txt "$rootdir"/ci/cppcheck/cppcheck_existing_issues.txt
```

Comment:
Why move it instead of directly generating it in the right location?

Actually, why move it on top of the existing one and use git diff? You could just as well leave it in the root directory and use normal diff. And if you want to avoid needing to install normal diff, you can use git diff with two files as arguments even if one of them is not tracked.